src/components/screens/content-editor-screen.tsx

```diff
@@ -1102,24 +1146,66 @@ class ContentEditorScreen extends React.Component<Props, State> {
       visitorVariablesApi.listVisitorVariables({ exhibitionId: exhibitionId })
     ]);
 
+    /**
+     * Devices from device group
+     */
     const deviceGroupId = groupContentVersion.deviceGroupId;
-    const [ devices, groupContentVersions ] = await Promise.all([
-      exhibitionDevicesApi.listExhibitionDevices({ exhibitionId, exhibitionGroupId: deviceGroupId }),
-      groupContentVersionsApi.listGroupContentVersions({ exhibitionId, deviceGroupId })
-    ]);
+    const devices = await exhibitionDevicesApi.listExhibitionDevices({ exhibitionId, exhibitionGroupId: deviceGroupId });
 
+    /**
+     * Content versions for different languages.
+     * Identified by having the same name but different language than
+     * initial content version selected before navigating to content editor.
+     */
     const contentVersions: ContentVersion[] = [];
-    for (const gContentVersion of groupContentVersions) {
-      const { contentVersionId } = gContentVersion;
-      if (!contentVersions.find(contentVersion => contentVersion.id === contentVersionId)) {
-        const contentVersion = await contentVersionsApi.findContentVersion({ exhibitionId, contentVersionId });
-        contentVersions.push(contentVersion);
+    contentVersionsInRoom.forEach(contentVersionInRoom => {
+      const sameName = contentVersionInRoom.name === initialContentVersion.name;
+      const newLanguage = contentVersions.every(version => version.language !== contentVersionInRoom.language);
+      if (sameName && newLanguage) {
+        contentVersions.push(contentVersionInRoom);
+      }
+    });
+
+    /**
+     * Group content versions for different content versions
+     * Creates new group content version for a content version if one does not exist yet
+     */
+    const groupContentVersions: GroupContentVersion[] = [];
+    for (const contentVersion of contentVersions) {
+      const foundGroupContentVersions = await groupContentVersionsApi.listGroupContentVersions({
+        exhibitionId,
+        contentVersionId: contentVersion.id!,
+        deviceGroupId: deviceGroupId
+      });
+
+      if (foundGroupContentVersions.length < 1) {
+        console.log(`No group content version found for content version "${contentVersion.name}", creating one now.`);
+        const newGroupContentVersion = await groupContentVersionsApi.createGroupContentVersion({
+          exhibitionId,
+          groupContentVersion: {
+            contentVersionId: contentVersion.id!,
+            deviceGroupId: deviceGroupId,
+            name: groupContentVersion.name,
+            status: GroupContentVersionStatus.Inprogress,
+            exhibitionId: exhibitionId
+          }
+        });
+
+        groupContentVersions.push(newGroupContentVersion);
+      } else {
+        if (foundGroupContentVersions.length > 1) {
+          console.error("More than one group content version with same device group found under one content version. This should not be possible!");
```

Comment:
Should this return if this case should not be possible?

Reply:
No need, as it takes the first one from the list anyways